public/app/core/components/OptionsUI/slider.tsx

```diff
@@ -1,26 +1,141 @@
+
+  // Using input font and expected maximum number of digits, set input width
+  useEffect(() => {
+    const inputElement = getComputedStyle(inputRef.current!);
+    const fontWeight = inputElement.getPropertyValue('font-weight') || 'normal';
+    const fontSize = inputElement.getPropertyValue('font-size') || '16px';
+    const fontFamily = inputElement.getPropertyValue('font-family') || 'Arial';
+    const marginDigits = 4; // extra digits to account for things like negative, exponential, and controls
+    const inputPadding = 8; // TODO: base this on input styling
+    const maxDigits =
+      Math.max((max + (step || 0)).toString().length, (max - (step || 0)).toString().length) + marginDigits;
+    const refString = '0'.repeat(maxDigits); // use "0" as widest numerical character
+    const calculatedTextWidth = getTextWidth(refString, `${fontWeight} ${fontSize} ${fontFamily}`);
+    if (calculatedTextWidth) {
+      setInputWidth(calculatedTextWidth + inputPadding * 2);
+    }
+  }, [max, step]);
+
+  const onSliderChange = useCallback(
+    (v: number) => {
+      setSliderValue(v);
+
+      if (onChange) {
+        onChange(v);
+      }
+    },
+    [setSliderValue, onChange]
+  );
+
+  const onSliderInputChange = useCallback(
+    (value?: number) => {
+      let v = value;
+
+      if (Number.isNaN(v) || !v) {
+        v = 0;
+      }
+
+      setSliderValue(v);
+
+      if (onChange) {
+        onChange(v);
+      }
+    },
+    [onChange]
+  );
+
+  const sliderInputClassNames = !isHorizontal ? [styles.sliderInputVertical] : [];
 
   return (
-    <Slider
-      value={initialValue}
-      min={settings?.min || 0}
-      max={settings?.max || 100}
-      step={settings?.step}
-      marks={settings?.marks}
-      included={settings?.included}
-      onChange={onChange}
-      ariaLabelForHandle={settings?.ariaLabelForHandle}
-    />
+    <div className={cx(styles.container, styles.slider)}>
+      {/** Slider tooltip's parent component is body and therefore we need Global component to do css overrides for it. */}
+      <Global styles={styles.slider} />
+      <label className={cx(styles.sliderInput, ...sliderInputClassNames)}>
+        <SliderWithTooltip
+          min={min}
+          max={max}
+          step={step}
+          defaultValue={value}
+          value={sliderValue}
+          onChange={onSliderChange}
+          vertical={!isHorizontal}
+          reverse={false}
+          ariaLabelForHandle={ariaLabelForHandle}
+          marks={marks}
+          included={included}
+        />
+        {/* Uses text input so that the number spinners are not shown */}
+        <span className={stylesSlider.numberInputWrapper} ref={inputRef}>
+          <NumberInput value={sliderValue} onChange={onSliderInputChange} max={max} min={min} step={step} />
+        </span>
+      </label>
+    </div>
   );
 };
+
+// Calculate width of string with given font
+function getTextWidth(text: string, font: string): number | null {
```

Comment:
is `measureText` not good enough?

Reply:
I am using that below on line 123, but it requires context to return the TextMetrics object.